From ca4be5f5073892d9a0a695b57b84a9522d41cb19 Mon Sep 17 00:00:00 2001
From: "ioana.ciornei" <ioana.ciornei@freescale.com>
Date: Wed, 15 Jul 2015 12:43:39 +0300
Subject: [PATCH 055/146] dpaa_eth: remove VERSION and dpa_get_ringparam

Removed VERSION parameter and get_ringparam callback since there is
no need for this functionality.

Signed-off-by: Ioana Ciornei <ioana.ciornei@freescale.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_debugfs.c  |  2 +-
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c      |  2 +-
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_base.c |  2 +-
 .../net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c  |  2 +-
 .../net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c  |  2 +-
 .../net/ethernet/freescale/sdk_dpaa/dpaa_eth_proxy.c    |  2 +-
 .../net/ethernet/freescale/sdk_dpaa/dpaa_eth_shared.c   |  2 +-
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ethtool.c  | 17 -----------------
 .../ethernet/freescale/sdk_dpaa/dpaa_generic_debugfs.c  |  2 +-
 .../ethernet/freescale/sdk_dpaa/dpaa_generic_ethtool.c  | 17 -----------------
 drivers/net/ethernet/freescale/sdk_dpaa/mac.c           |  3 +--
 drivers/net/ethernet/freescale/sdk_dpaa/offline_port.c  |  2 +-
 12 files changed, 10 insertions(+), 45 deletions(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_debugfs.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_debugfs.c
index 59c7338..cb070f5 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_debugfs.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_debugfs.c
@@ -366,7 +366,7 @@ int __init dpa_debugfs_module_init(void)
 {
 	int	 _errno = 0;
 
-	pr_info(KBUILD_MODNAME ": " DPA_DEBUGFS_DESCRIPTION " (" VERSION ")\n");
+	pr_info(KBUILD_MODNAME ": " DPA_DEBUGFS_DESCRIPTION "\n");
 
 	dpa_debugfs_root = debugfs_create_dir(DPA_ETH_DEBUGFS_ROOT, NULL);
 
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c
index a5fb44f..92e697e 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.c
@@ -1131,7 +1131,7 @@ static int __init __cold dpa_load(void)
 {
 	int	 _errno;
 
-	pr_info(DPA_DESCRIPTION " (" VERSION ")\n");
+	pr_info(DPA_DESCRIPTION "\n");
 
 #ifdef CONFIG_FSL_DPAA_ETH_DEBUGFS
 	dpa_debugfs_module_init();
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_base.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_base.c
index 9f55496..459823e 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_base.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_base.c
@@ -235,7 +235,7 @@ EXPORT_SYMBOL(dpa_bp_create);
 
 static int __init __cold dpa_advanced_load(void)
 {
-	pr_info(DPA_DESCRIPTION " (" VERSION ")\n");
+	pr_info(DPA_DESCRIPTION "\n");
 
 	return 0;
 }
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c
index 37effc1..f672420 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_generic.c
@@ -1548,7 +1548,7 @@ static int __init __cold dpa_generic_load(void)
 {
 	int	 _errno;
 
-	pr_info(KBUILD_MODNAME ": " DPA_GENERIC_DESCRIPTION " (" VERSION ")\n");
+	pr_info(KBUILD_MODNAME ": " DPA_GENERIC_DESCRIPTION "\n");
 
 #ifdef CONFIG_FSL_DPAA_ETH_DEBUGFS
 	dpa_generic_debugfs_module_init();
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c
index bd97536..68b6e5b 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_macless.c
@@ -468,7 +468,7 @@ static int __init __cold dpa_macless_load(void)
 {
 	int	 _errno;
 
-	pr_info(DPA_DESCRIPTION " (" VERSION ")\n");
+	pr_info(DPA_DESCRIPTION "\n");
 
 	/* Initialize dpaa_eth mirror values */
 	dpa_rx_extra_headroom = fm_get_rx_extra_headroom();
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_proxy.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_proxy.c
index 8ad015d..994d38c 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_proxy.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_proxy.c
@@ -351,7 +351,7 @@ static int __init __cold dpa_proxy_load(void)
 {
 	int	 _errno;
 
-	pr_info(DPA_DESCRIPTION " (" VERSION ")\n");
+	pr_info(DPA_DESCRIPTION "\n");
 
 	/* Initialize dpaa_eth mirror values */
 	dpa_rx_extra_headroom = fm_get_rx_extra_headroom();
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_shared.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_shared.c
index 987b9dc..6564758 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_shared.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth_shared.c
@@ -880,7 +880,7 @@ static int __init __cold dpa_shared_load(void)
 {
 	int	 _errno;
 
-	pr_info(DPA_DESCRIPTION " (" VERSION ")\n");
+	pr_info(DPA_DESCRIPTION "\n");
 
 	/* Initialize dpaa_eth mirror values */
 	dpa_rx_extra_headroom = fm_get_rx_extra_headroom();
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ethtool.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ethtool.c
index 60a1b22..68daa39 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ethtool.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ethtool.c
@@ -99,8 +99,6 @@ static void __cold dpa_get_drvinfo(struct net_device *net_dev,
 
 	strncpy(drvinfo->driver, KBUILD_MODNAME,
 		sizeof(drvinfo->driver) - 1)[sizeof(drvinfo->driver)-1] = 0;
-	strncpy(drvinfo->version, VERSION,
-		sizeof(drvinfo->driver) - 1)[sizeof(drvinfo->version)-1] = 0;
 	_errno = snprintf(drvinfo->fw_version, sizeof(drvinfo->fw_version),
 			  "%X", 0);
 
@@ -153,20 +151,6 @@ static int __cold dpa_nway_reset(struct net_device *net_dev)
 	return _errno;
 }
 
-static void __cold dpa_get_ringparam(struct net_device *net_dev,
-		struct ethtool_ringparam *et_ringparam)
-{
-	et_ringparam->rx_max_pending	   = 0;
-	et_ringparam->rx_mini_max_pending  = 0;
-	et_ringparam->rx_jumbo_max_pending = 0;
-	et_ringparam->tx_max_pending	   = 0;
-
-	et_ringparam->rx_pending	   = 0;
-	et_ringparam->rx_mini_pending	   = 0;
-	et_ringparam->rx_jumbo_pending	   = 0;
-	et_ringparam->tx_pending	   = 0;
-}
-
 static void __cold dpa_get_pauseparam(struct net_device *net_dev,
 		struct ethtool_pauseparam *epause)
 {
@@ -320,7 +304,6 @@ const struct ethtool_ops dpa_ethtool_ops = {
 	.get_msglevel = dpa_get_msglevel,
 	.set_msglevel = dpa_set_msglevel,
 	.nway_reset = dpa_nway_reset,
-	.get_ringparam = dpa_get_ringparam,
 	.get_pauseparam = dpa_get_pauseparam,
 	.set_pauseparam = dpa_set_pauseparam,
 	.self_test = NULL, /* TODO invoke the cold-boot unit-test? */
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_debugfs.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_debugfs.c
index b53c289..a442876 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_debugfs.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_debugfs.c
@@ -256,7 +256,7 @@ int __init dpa_generic_debugfs_module_init(void)
 {
 	int	 _errno = 0;
 
-	pr_info(KBUILD_MODNAME ": " DPA_DEBUGFS_DESCRIPTION " (" VERSION ")\n");
+	pr_info(KBUILD_MODNAME ": " DPA_DEBUGFS_DESCRIPTION "\n");
 
 	dpa_debugfs_root = debugfs_create_dir(DPA_GENERIC_ETH_DEBUGFS_ROOT,
 			NULL);
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_ethtool.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_ethtool.c
index b9c8623..ef41ce3 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_ethtool.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_generic_ethtool.c
@@ -65,8 +65,6 @@ static void __cold dpa_generic_get_drvinfo(struct net_device *net_dev,
 
 	strncpy(drvinfo->driver, KBUILD_MODNAME,
 		sizeof(drvinfo->driver) - 1)[sizeof(drvinfo->driver)-1] = 0;
-	strncpy(drvinfo->version, VERSION,
-		sizeof(drvinfo->driver) - 1)[sizeof(drvinfo->version)-1] = 0;
 	_errno = snprintf(drvinfo->fw_version, sizeof(drvinfo->fw_version),
 			  "%X", 0);
 
@@ -99,20 +97,6 @@ static int __cold dpa_generic_nway_reset(struct net_device *net_dev)
 	return -ENODEV;
 }
 
-static void __cold dpa_generic_get_ringparam(struct net_device *net_dev,
-		struct ethtool_ringparam *et_ringparam)
-{
-	et_ringparam->rx_max_pending	   = 0;
-	et_ringparam->rx_mini_max_pending  = 0;
-	et_ringparam->rx_jumbo_max_pending = 0;
-	et_ringparam->tx_max_pending	   = 0;
-
-	et_ringparam->rx_pending	   = 0;
-	et_ringparam->rx_mini_pending	   = 0;
-	et_ringparam->rx_jumbo_pending	   = 0;
-	et_ringparam->tx_pending	   = 0;
-}
-
 const struct ethtool_ops dpa_generic_ethtool_ops = {
 	.get_settings = dpa_generic_get_settings,
 	.set_settings = dpa_generic_set_settings,
@@ -120,6 +104,5 @@ const struct ethtool_ops dpa_generic_ethtool_ops = {
 	.get_msglevel = dpa_generic_get_msglevel,
 	.set_msglevel = dpa_generic_set_msglevel,
 	.nway_reset = dpa_generic_nway_reset,
-	.get_ringparam = dpa_generic_get_ringparam,
 	.get_link = ethtool_op_get_link,
 };
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/mac.c b/drivers/net/ethernet/freescale/sdk_dpaa/mac.c
index f36d091..d004038 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/mac.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/mac.c
@@ -436,8 +436,7 @@ static int __init __cold mac_load(void)
 	pr_debug(KBUILD_MODNAME ": -> %s:%s()\n",
 		KBUILD_BASENAME".c", __func__);
 
-	pr_info(KBUILD_MODNAME ": %s (" VERSION ")\n",
-		mac_driver_description);
+	pr_info(KBUILD_MODNAME ": %s\n", mac_driver_description);
 
 	_errno = platform_driver_register(&mac_driver);
 	if (unlikely(_errno < 0)) {
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/offline_port.c b/drivers/net/ethernet/freescale/sdk_dpaa/offline_port.c
index ddbfdce..fb084af 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/offline_port.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/offline_port.c
@@ -820,7 +820,7 @@ static int __init __cold oh_port_load(void)
 {
 	int _errno;
 
-	pr_info(OH_MOD_DESCRIPTION " (" VERSION ")\n");
+	pr_info(OH_MOD_DESCRIPTION "\n");
 
 	_errno = platform_driver_register(&oh_port_driver);
 	if (_errno < 0) {
-- 
2.1.0.27.g96db324

